core/src/main/java/org/apache/spark/util/collection/unsafe/sort/UnsafeExternalSorter.java

```diff
@@ -113,7 +151,6 @@ public BoxedUnit apply() {
    * Allocates new sort data structures. Called when creating the sorter and after each spill.
    */
   private void initializeForWriting() throws IOException {
-    this.writeMetrics = new ShuffleWriteMetrics();
```

Comment:
this was moved into the ctor, as I wasn't sure we wanted to reset writeMetrics after every spill.